### PR TITLE
[SYCLomatic] Add examples cuBLAS API migration

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/examples/DPCT/cuBLAS/cublasDasum.cu [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include "cublas_v2.h"
+
+void test(cublasHandle_t h, int n, const double *x, int incx, double *res) {
+  // Start
+  cublasDasum(h /*cublasHandle_t*/, n /*int*/, x /*const double **/,
+              incx /*int*/, res /*double* */);
```
Comment (zhiweij1) [inline clang/examples/DPCT/cuBLAS/cublasDasum.cu]:
```suggestion
              incx /*int*/, res /*double **/);
```

File: clang/examples/DPCT/cuBLAS/cublasDzasum.cu [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#include "cublas_v2.h"
+
+void test(cublasHandle_t h, int n, const cuDoubleComplex *x, int incx,
+          double *res) {
+  // Start
+  cublasDzasum(h /*cublasHandle_t*/, n /*int*/, x /*const cuDoubleComplex **/,
+               incx /*int*/, res /*double* */);
```
Comment (zhiweij1) [inline clang/examples/DPCT/cuBLAS/cublasDzasum.cu]:
```suggestion
               incx /*int*/, res /*double **/);
```

File: clang/examples/DPCT/cuBLAS/cublasSasum.cu [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#include "cublas_v2.h"
+
+void test(cublasHandle_t h, int n, const float *x, int incx, float *res) {
+  // Start
+  cublasSasum(h /*cublasHandle_t*/, n /*int*/, x /*const float **/,
+              incx /*int*/, res /*float* */);
```
Comment (zhiweij1) [inline clang/examples/DPCT/cuBLAS/cublasSasum.cu]:
```suggestion
              incx /*int*/, res /*float **/);
```

File: clang/examples/DPCT/cuBLAS/cublasScasum.cu [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#include "cublas_v2.h"
+
+void test(cublasHandle_t h, int n, const cuComplex *x, int incx, float *res) {
+  // Start
+  cublasScasum(h /*cublasHandle_t*/, n /*int*/, x /*const cuComplex **/,
+               incx /*int*/, res /*float* */);
```
Comment (zhiweij1) [inline clang/examples/DPCT/cuBLAS/cublasScasum.cu]:
```suggestion
               incx /*int*/, res /*float **/);
```

File: clang/examples/DPCT/cuBLAS/cublasAxpyEx.cu [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#include "cublas_v2.h"
```
Comment (tangjj11) [inline clang/examples/DPCT/cuBLAS/cublasAxpyEx.cu:1]: Please add lit test of each case.

**Review by tangjj11 — CHANGES_REQUESTED** [review]

File: clang/examples/DPCT/cuBLAS/cublasAxpyEx.cu (hunk 5 above)
Comment (zhiweij1) [reply clang/examples/DPCT/cuBLAS/cublasAxpyEx.cu:1]: done

**Review by tangjj11 — APPROVED** [review]

File: clang/test/dpct/query_api_mapping/cuBLAS/blas.cu [hunk 6]
```diff
@@ -0,0 +1,2045 @@
...
+// cublasSspmv-NEXT: Is migrated to:
+// cublasSspmv-NEXT:   oneapi::mkl::blas::column_major::spmv(*handle, upper_lower, n, dpct::get_value(alpha, *handle), a, x, incx, dpct::get_value(beta, *handle), y, incy);
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=cublasCsrot | FileCheck %s -check-prefix=cublasCsrot
+// cublasCsrot: CUDA API:
+// cublasCsrot-NEXT:   cublasCsrot(handle /*cublasHandle_t*/, n /*int*/, x /*cuComplex **/,
+// cublasCsrot-NEXT:               incx /*int*/, y /*cuComplex **/, incy /*int*/,
+// cublasCsrot-NEXT:               c /*const float **/, s /*const float **/);
+// cublasCsrot-NEXT: Is migrated to:
+// cublasCsrot-NEXT:   oneapi::mkl::blas::column_major::rot(*handle, n, (std::complex<float>*)x, incx, (std::complex<float>*)y, incy, dpct::get_value(c, *handle), dpct::get_value(s, *handle));
```
Comment (zhimingwang36) [inline clang/test/dpct/query_api_mapping/cuBLAS/blas.cu]: let's split the file to speed up the lit test.

File: clang/test/dpct/query_api_mapping/cuBLAS/blas_9.cu [hunk 7]
```diff
@@ -0,0 +1,254 @@
+// UNSUPPORTED: cuda-8.0, cuda-9.0, cuda-9.1
```
Comment (zhimingwang36) [inline clang/test/dpct/query_api_mapping/cuBLAS/blas_9.cu]: change file name to blas_9.2.cu

File: clang/test/dpct/query_api_mapping/cuBLAS/blas.cu (hunk 6 above)
Comment (zhiweij1) [reply clang/test/dpct/query_api_mapping/cuBLAS/blas.cu]: done

File: clang/test/dpct/query_api_mapping/cuBLAS/blas_9.cu (hunk 7 above)
Comment (zhiweij1) [reply clang/test/dpct/query_api_mapping/cuBLAS/blas_9.cu]: changed to blas_9_2.cu

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**